src/core/network/createNetwork.js

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import createPayload from "./createPayload";
+import createResponse from "./createResponse";
+import defer from "../../utils/defer";
+
+export default (config, logger, lifecycle, networkStrategy) => {
+  const handleResponse = ({ body, promise }) => {
```

Comment:
This would come from the network strategy.  One of the features of Konductor will be its support for HTTP 2.0 streaming responses, so if the fetch network strategy supported streaming responses it could return a promise for the next response.  I will add a comment here to explain this.

Reply:
This might end up falling under the YAGNI principle. I'm not sure if we're going to need it (choose to support streaming responses).